src/macros.rs

```diff
@@ -129,48 +134,104 @@
 /// // Doesn't compile because `foo` isn't a `Bar`
 /// doc! { field!((foo in MyModel).(baz in Bar)): 0 };
 /// ```
+/// ```compile_fail
+///# use mongodm::mongo::bson::doc;
+///# use mongodm::field;
+///#
+///# struct MyModel {
+///#     foo: i64,
+///#     bar: Bar,
+///# }
+///#
+///# struct Bar {
+///#     baz: i64,
+///#     third: Third,
+///# }
+///#
+///# struct Third {
+///#     a: String,
+///# }
+///#
+/// // Fail because `b` is not a field of `Third`
+/// doc! { field!((bar in MyModel).(third in Bar).(b in Third)): 0 };
+/// ```
 #[macro_export]
 macro_rules! field {
-    ( $field:ident in $type:path ) => {{
-        #[allow(unknown_lints, unneeded_field_pattern)]
-        const _: fn() = || {
-            let $type { $field: _, .. };
-        };
-        stringify!( $field )
+    ( $($tt:tt)* ) => {{
+        $crate::field_check_helper! { $($tt)* }
+        $crate::field_string_helper! { $($tt)* }
     }};
-    ( @ $field:ident in $type:path ) => {{
+}
+
+#[macro_export]
```

Comment:
```suggestion
#[doc(hide)]
#[macro_export]
```
Since this isn't supposed to be used directly.

Reply:
100% agree, fixing it.